src/interface/misc.rs

```diff
@@ -165,3 +166,91 @@ impl AdvisoryLock {
         } else {false}
     }
 }
+
+pub struct RawMutex {
+    inner: libc::pthread_mutex_t
+}
+
+impl RawMutex {
+    pub fn create() -> Result<Self, i32> {
+        let inner;
+        let libcret;
+        unsafe {
+            inner = std::mem::MaybeUninit::uninit();
+            libcret = libc::pthread_mutex_init((&mut inner.assume_init()) as *mut libc::pthread_mutex_t, std::ptr::null());
+        }
+        if libcret < 0 { Err(libcret) } else { Ok(Self {inner: unsafe{inner.assume_init()}}) }
+    }
+    pub fn lock(&self) -> i32 {
```

Comment:
Its a bit easier to read if you separate these functions with a newline between them.